Build fails because of JOOQ 3.13.0

[JOOQ](https://github.com/jOOQ/jOOQ) released version [3.13.0 on February 12th 2020](https://github.com/jOOQ/jOOQ/releases/tag/version-3.13.0).

Micrometer is incompatible with this release, and because it pins the version to [latest.release](https://github.com/micrometer-metrics/micrometer/blob/f6a2687f13f2b9fd7351ce9126a9f04238da79dd/dependencies.gradle#L55), this has broken the build:

```
$ ./gradlew clean compileJava
... snip ...
> Task :micrometer-core:compileJava FAILED
/Users/0000000/git/micrometer-metrics/micrometer/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/db/MetricsDSLContext.java:63: error: MetricsDSLContext is not abstract and does not override abstract method dropTemporaryTableIfExists(Table<?>) in DSLContext
public class MetricsDSLContext implements DSLContext {
       ^
... snip ...

20 errors
47 warnings

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':micrometer-core:compileJava'.
> Compilation failed; see the compiler error output for details.
```

--- a/dependencies.gradle
+++ b/dependencies.gradle
@@ -52,7 +52,8 @@ def VERSIONS = [
         'org.hdrhistogram:HdrHistogram:latest.release',
         'org.hibernate:hibernate-entitymanager:latest.release',
         'org.hsqldb:hsqldb:latest.release',
-        'org.jooq:jooq:latest.release',
+        // Pin version temporarily to restore builds. See gh-1852
+        'org.jooq:jooq:3.12.+',
         'org.jsr107.ri:cache-ri-impl:1.0.0',
         'org.junit.jupiter:junit-jupiter:latest.release',
         'org.junit.platform:junit-platform-launcher:latest.release',

